### PR TITLE
[FEATURE] Implement document validation before document is saved to SOLR

### PR DESCRIPTION
**Pull request by beatrycze-volk**
There exists possibility that during save to database document contains 'record_id', but already during saving to SOLR it doesn't. It is necessary to validate fields in both places to avoid it.

Depends on #1149

### CODE REVIEW

**Comment by beatrycze-volk** [comment]
Documentation build failed:

```
In inline-node.html.twig line 2:
                                                                               
  An exception has been thrown during the rendering of a template ("Failed to  
   connect to docs.typo3.org port 80 after 2010 ms: Couldn't connect to serve  
  r for "http://docs.typo3.org/typo3cms/TyposcriptReference/objects.inv.json"  
  .").                                                                         
                                                                               

In ErrorChunk.php line 56:
                                                                               
  Failed to connect to docs.typo3.org port [80](https://github.com/kitodo/kitodo-presentation/actions/runs/9269551905/job/25500591891?pr=1158#step:3:81) after 2010 ms: Couldn't connect  
   to server for "http://docs.typo3.org/typo3cms/TyposcriptReference/objects.  
  inv.json".                                                                   
                                                                               

In CurlResponse.php line 315:
                                                                               
  Failed to connect to docs.typo3.org port 80 after 2010 ms: Couldn't connect  
   to server for "http://docs.typo3.org/typo3cms/TyposcriptReference/objects.  
  inv.json".
```

File: Classes/Command/IndexCommand.php [hunk 1]
```diff
@@ -194,13 +194,19 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 if ($io->isVerbose()) {
                     $io->section('Indexing ' . $document->getUid() . ' ("' . $document->getLocation() . '") on Solr core ' . $solrCoreUid . '.');
                 }
-                Indexer::add($document, $this->documentRepository);
+                $isSaved = Indexer::add($document, $this->documentRepository);
+            } else {
+                $io->error('ERROR: Document with UID "' . $document->getUid() . '" could not be indexed on PID ' . $this->storagePid . ' . There are missing mandatory fields (at least one of those: ' . $this->extConf['requiredMetadataFields'] .') in this document.');
```
Comment (sebastian-meyer) [inline Classes/Command/IndexCommand.php]:
```suggestion
                $io->error('ERROR: Document with UID "' . $document->getUid() . '" could not be indexed on PID ' . $this->storagePid . ' . There are missing mandatory fields (at least one of those: ' . $this->extConf['requiredMetadataFields'] . ') in this document.');
```

File: Classes/Command/IndexCommand.php [hunk 2]
```diff
@@ -194,13 +194,19 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+                return BaseCommand::FAILURE;
+            }
 
+            if ($isSaved) {
                 $io->success('All done!');
                 return BaseCommand::SUCCESS;
             }
 
-            $io->error('ERROR: Document with UID "' . $document->getUid() . '" could not be indexed on PID ' . $this->storagePid . ' . There are missing mandatory fields (document format or record identifier) in this document.');
+            $io->error('ERROR: Document with UID "' . $document->getUid() . '" could not be indexed on Solr core ' . $solrCoreUid . ' . There are missing mandatory fields (at least one of those: ' . $this->extConf['requiredMetadataFields'] .') in this document.');
```
Comment (sebastian-meyer) [inline Classes/Command/IndexCommand.php]:
```suggestion
            $io->error('ERROR: Document with UID "' . $document->getUid() . '" could not be indexed on Solr core ' . $solrCoreUid . ' . There are missing mandatory fields (at least one of those: ' . $this->extConf['requiredMetadataFields'] . ') in this document.');
```

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]
Just a few Codacy issues...

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**